File: clean.py
import re

file_path = r'c:\Users\LENOVO\OneDrive\سطح المكتب\Test-1\src\app\features\properties\property-list\property-list.ts'
content = open(file_path, encoding='utf-8').read()

# Let's find 'clearAllCompare() {' and remove everything after it up to the closing class brace.
idx = content.find('clearAllCompare() {')
if idx != -1:
    content = content[:idx] + '}\n'

open(file_path, 'w', encoding='utf-8').write(content)
print("SUCCESS")

File: clean.js
const fs = require('fs');
const path = './src/app/features/properties/saved-properties/saved-properties.ts';
let content = fs.readFileSync(path, 'utf8');

// We want to find the index of "<!-- Floating Comparison Tray -->"
const startIdx = content.indexOf('<!-- Floating Comparison Tray -->');
// We want to find the index of "    </div>\r\n  `," or "    </div>\n  `," at the very end of template
let endIdx = content.indexOf('    </div>\r\n  `,');
if (endIdx === -1) {
  endIdx = content.indexOf('    </div>\n  `,');
}

if (startIdx !== -1 && endIdx !== -1) {
  const replacement = '      <app-property-compare [selectedProperties]="selectedPropertiesForCompare()" (compareChange)="selectedPropertiesForCompare.set($event)" />\n';
  content = content.substring(0, startIdx) + replacement + content.substring(endIdx);
  fs.writeFileSync(path, content, 'utf8');
  console.log("SUCCESS TEMPLATE");
} else {
  console.log("TEMPLATE NOT FOUND", startIdx, endIdx);
}
